File: Part2/Rust/BasicConcepts/BasicConcepts_MixedTypes.rs
fn main()
{
    // Do not mix your types in many languages
    // Rust is strongly typed and will not allow implicit type conversions
    // These will cause compile errors due to type mismatches
    let my_int: i32 = 123.45;
    let my_float: f32 = 'a';
    let my_char: char = 543.21;
    
    println!("myInt: {}", my_int);
    println!("myFloat: {}", my_float);
    println!("myChar: {}", my_char);
}

File: Part2/Go/BasicConcepts/BasicConcepts_MixedTypes.go
package main

import "fmt"

func main() {
	// Do not mix your types in many languages
	// Go is strongly typed and will not allow implicit type conversions
	// These will cause compile errors
	var myInt int = 123.45
	var myFloat float32 = 'a'
	var myChar byte = 543.21

	fmt.Printf("myInt: %v\n", myInt)
	fmt.Printf("myFloat: %v\n", myFloat)
	fmt.Printf("myChar: %v\n", myChar)
}
